### PR TITLE
Test: 3rd party is reloaded with passphrase

### LINKED ISSUE #8775: Test: popup e2e tests with passphrase
We have some passphrase testing for popup but it would be nice to handle some of the edge cases:

```[tasklist]
### Tasks
- [x] 3rd party reloading with passphrase
- [x] passphrase missmatch
```

also try to check this one https://github.com/MetaMask/metamask-extension/issues/12967

### PR DESCRIPTION
**Pull request by karliatto**
## Description

Extend e2e tests for passphrase to address some of the situation mentioned https://github.com/trezor/trezor-suite/issues/8739  where 3rd party is reloaded and when it is not.

It also adds some helper function to make tests easier to write.

## Related Issue

Related https://github.com/trezor/trezor-suite/issues/8739 
Closes https://github.com/trezor/trezor-suite/issues/8775

### CODE REVIEW

File: packages/connect-popup/e2e/support/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-await-in-loop */
+
+import { Page } from '@playwright/test';
+
+// Waits and clicks for an array on buttons in serial order.
+export const waitAndClickButtons = async (page: Page, buttons: string[]) => {
```
Comment (mroz22) [inline packages/connect-popup/e2e/support/helpers.ts]: maybe `waitAndClick` or `waitAndClickElements` ? this makes one feel it need to be button elements

**Review by mroz22 — CHANGES_REQUESTED** [review]
also I pushed one fixup, I would like to have "passphrase mismatch" also covered, may I ask you to finish it please 🙏 
to learn more:
- https://docs.trezor.io/trezor-firmware/common/communication/passphrase.html
- https://docs.trezor.io/trezor-suite/packages/connect/methods/commonParams.html

File: packages/connect-popup/e2e/support/helpers.ts (hunk 1 above)
Comment (karliatto) [reply packages/connect-popup/e2e/support/helpers.ts]: :+1:  Changed it to `waitAndClick`

**Comment by karliatto** [comment]
> also I pushed one fixup, I would like to have "passphrase mismatch" also covered, may I ask you to finish it please pray to learn more:
> 
>     * https://docs.trezor.io/trezor-firmware/common/communication/passphrase.html
> 
>     * https://docs.trezor.io/trezor-suite/packages/connect/methods/commonParams.html

Thanks! I updated it to address the situations:
* getting error passphrase mismatch and accept the mismatched passphrase
* getting error passphrase mismatch and change it to the right one

**Review by mroz22 — APPROVED** [review]

**Pull request merged**